crates/solana-gateway-task-processor/src/component.rs

```diff
@@ -480,6 +504,49 @@ async fn gateway_tx_task(
     Ok(())
 }
 
+async fn refund_task(
+    task: amplifier_api::types::RefundTask,
+    metadata: &ConfigMetadata,
+    solana_rpc_client: &RpcClient,
+    keypair: &Keypair,
+) -> eyre::Result<()> {
+    let receiver = Pubkey::from_str(&task.refund_recipient_address)?;
+    let mut message_id_parts = task.message.message_id.0.split('-');
+    let tx_hash = Signature::from_str(message_id_parts.next().ok_or_eyre("missing tx hash")?)?
+        .as_ref()
+        .try_into()?;
+    let log_index = message_id_parts
+        .next()
+        .ok_or_eyre("missing log_index")?
+        .parse()?;
+
+    if task.remaining_gas_balance.token_id.is_some() {
+        // Should we implement this? As the Gas Service supports it, perhaps we should.
+        //
+        // We need a function to build the SPL refund instruction before we implement this
+        // though.
+        tracing::warn!("ignoring refund task for non-native token");
```

Comment:
I will change this to `eyre::bail!`. Because Axelar does not support non-native gas payments. And I don't want to just silently ignore them. I'll see if I can report back somehow that we cannot execute this too